Realize method

YuliaTsariv opened the pull request.

ChumakAn commented on pom.xml:
```diff
@@ -51,6 +51,14 @@
+                    <artifactId>maven-surefire-plugin</artifactId>
+                    <version>2.19.1</version>
+                    <configuration>
+                        <testFailureIgnore>true</testFailureIgnore>
```
don't let ignore test failure

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,51 @@
+        List<String[]> list = new ArrayList<>();
+        List<String> listOfPosition = new ArrayList<>();
+        int supplyResult = 0;
+        try (
+                BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFileName));
+                BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName,
+                        true));) {
+            while (bufferedReader.ready()) {
```
try to format your code due to code style

ChumakAn commented on src/test/java/core/basesyntax/WorkWithFileTest.java:
```diff
@@ -60,7 +60,7 @@ public void getStatisticAboutGrape() {
 
     @Test
-    public void getStatisticAboutBanana() {
+    public void getStatisticAboutBanana() throws IOException {
```
don't change tests

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,51 @@
+                        result += Integer.parseInt(str[1]);
+                    }
+                }
+                bufferedWriter.write(lst + "," + result + "\r\n");
```
these signs can be constants

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,51 @@
+                list.add(rows);
+            }
+            for (String[] str : list) {
+                if (!listOfPosition.contains(str[0])) {
```
the indexes can be constants

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,51 @@
+public class WorkWithFile {
 
+    public void getStatistic(String fromFileName, String toFileName) throws IOException {
+        List<String[]> list = new ArrayList<>();
```
why do you need list of strings?

ChumakAn reviewed (changes requested):
Try to break your one huge method to some smaller ones and use the smaller methods in getStatistic method. It will help you to make your code more readable

ChumakAn commented on src/main/java/core/basesyntax/Test.java:
```diff
@@ -0,0 +1,34 @@
+package core.basesyntax;
+
+public class Test {
+    public static void main(String[] args) {
+        System.out.println(extractFileName("0_file.txt.zip"));
+    }
+
+    public static String extractFileName(String dirtyFileName) {
+        String[] str = dirtyFileName.split("_");
+        StringBuilder stringBuilder = new StringBuilder();
+        System.out.println("string " + str[0] + " " + str[1]);
+        if (str.length > 2) {
+            System.out.println("mskmi  " + str.length);
+            for (int a = 1; a < str.length; a++) {
+                stringBuilder.append(str[a]).append("_");
+            }
+        } else {
+            stringBuilder.append(str[1]);
+        }
+
+        String[] strWithDots = stringBuilder.toString().split("[.]");
+        System.out.println("strWdots " + strWithDots[0]);
+        for (String test : strWithDots) {
+            System.out.println("test " + test);
+        }
+        StringBuilder stringBuilder1 = new StringBuilder();
+        for (int k = 0; k < strWithDots.length - 1; k++) {
+            stringBuilder1.append(strWithDots[k]).append(".");
+        }
+        System.out.println("String " + stringBuilder1);
+        stringBuilder1.deleteCharAt(stringBuilder1.length() - 1);
+        return stringBuilder1.toString();
+    }
+}
```
what's the purpose of the class?

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,66 @@
+    private static final int INDEX_AMMOUNT = 1;
+
     public void getStatistic(String fromFileName, String toFileName) {
+        List<String> listOfPosition = new ArrayList<>();
+        List<String[]> list = getDataFromFile(fromFileName);
+        int supplyResult = 0;
+
+        try (BufferedWriter bufferedWriter =
+                     new BufferedWriter(new FileWriter(toFileName, true))) {
+            listOfPosition.add(SUPPLY);
+            listOfPosition.add(BUY);
+            for (String lst : listOfPosition) {
+                int result = 0;
+                for (String[] str : list) {
+                    if (str[INDEX_TYPE_OF_OPERATION].equals(lst)) {
+                        result += Integer.parseInt(str[INDEX_AMMOUNT]);
+                    }
+                }
+                bufferedWriter.write(lst + COMMA + result + ENTER);
+                if (supplyResult == 0) {
+                    supplyResult = result;
+                } else {
+                    int difference = supplyResult - result;
+                    bufferedWriter.write("result," + difference);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
```
this method is too large, try to break it in smaller ones

ChumakAn commented on src/test/java/core/basesyntax/WorkWithFileTest.java:
```diff
@@ -27,7 +27,7 @@ public void clearResults() {
 
     @Test
-    public void getStatisticAboutApple() {
+    public void getStatisticAboutApple()  {
```
don't edit anything in WorkWithFileTest class

ChumakAn reviewed (changes requested):
Let's improve your solution, also take a notice at Java CI build failure, you can find the detailed report, and fix this.

YuliaTsariv commented:
Добрий день
виконала
ср, 5 жовт. 2022 р. о 12:48 Anastasiia Chumak ***@***.***>
пише:

> ***@***.**** requested changes on this pull request.
>
> Let's improve your solution, also take a notice at Java CI build failure,
> you can find the detailed report, and fix this.
> ------------------------------
>
> In src/main/java/core/basesyntax/Test.java
> <https://github.com/mate-academy/jv-work-with-file/pull/601#discussion_r987735859>
> :
>
> > +package core.basesyntax;
> +
> +public class Test {
> +    public static void main(String[] args) {
> +        System.out.println(extractFileName("0_file.txt.zip"));
> +    }
> +
> +    public static String extractFileName(String dirtyFileName) {
> +        String[] str = dirtyFileName.split("_");
> +        StringBuilder stringBuilder = new StringBuilder();
> +        System.out.println("string " + str[0] + " " + str[1]);
> +        if (str.length > 2) {
> +            System.out.println("mskmi  " + str.length);
> +            for (int a = 1; a < str.length; a++) {
> +                stringBuilder.append(str[a]).append("_");
> +            }
> +        } else {
> +            stringBuilder.append(str[1]);
> +        }
> +
> +        String[] strWithDots = stringBuilder.toString().split("[.]");
> +        System.out.println("strWdots " + strWithDots[0]);
> +        for (String test : strWithDots) {
> +            System.out.println("test " + test);
> +        }
> +        StringBuilder stringBuilder1 = new StringBuilder();
> +        for (int k = 0; k < strWithDots.length - 1; k++) {
> +            stringBuilder1.append(strWithDots[k]).append(".");
> +        }
> +        System.out.println("String " + stringBuilder1);
> +        stringBuilder1.deleteCharAt(stringBuilder1.length() - 1);
> +        return stringBuilder1.toString();
> +    }
> +}
>
> what's the purpose of the class?
> ------------------------------
>
> In src/main/java/core/basesyntax/WorkWithFile.java
> <https://github.com/mate-academy/jv-work-with-file/pull/601#discussion_r987736299>
> :
>
> > +        List<String> listOfPosition = new ArrayList<>();
> +        List<String[]> list = getDataFromFile(fromFileName);
> +        int supplyResult = 0;
> +
> +        try (BufferedWriter bufferedWriter =
> +                     new BufferedWriter(new FileWriter(toFileName, true))) {
> +            listOfPosition.add(SUPPLY);
> +            listOfPosition.add(BUY);
> +            for (String lst : listOfPosition) {
> +                int result = 0;
> +                for (String[] str : list) {
> +                    if (str[INDEX_TYPE_OF_OPERATION].equals(lst)) {
> +                        result += Integer.parseInt(str[INDEX_AMMOUNT]);
> +                    }
> +                }
> +                bufferedWriter.write(lst + COMMA + result + ENTER);
> +                if (supplyResult == 0) {
> +                    supplyResult = result;
> +                } else {
> +                    int difference = supplyResult - result;
> +                    bufferedWriter.write("result," + difference);
> +                }
> +            }
> +        } catch (IOException e) {
> +            e.printStackTrace();
> +        }
>
> this method is too large, try to break it in smaller ones
> ------------------------------
>
> In src/test/java/core/basesyntax/WorkWithFileTest.java
> <https://github.com/mate-academy/jv-work-with-file/pull/601#discussion_r987736683>
> :
>
> > @@ -27,7 +27,7 @@ public void clearResults() {
>      }
>
>      @Test
> -    public void getStatisticAboutApple() {
> +    public void getStatisticAboutApple()  {
>
> don't edit anything in WorkWithFileTest class
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/mate-academy/jv-work-with-file/pull/601#pullrequestreview-1131165643>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/A26KY4CVX3S5HX2A5HNQZE3WBVFHJANCNFSM6AAAAAAQ3T5BGY>
> .
> You are receiving this because you authored the thread.Message ID:
> ***@***.***>
>

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,72 @@
+        List<String[]> listOfData = getDataFromFile(fromFileName);
+        int supplyResult = 0;
+
+        try (BufferedWriter bufferedWriter =
+                     new BufferedWriter(new FileWriter(toFileName, true))) {
+            listOfPosition.add(SUPPLY);
+            listOfPosition.add(BUY);
+            for (String lst : listOfPosition) {
+                int result = countTheResult(listOfData, lst);
+                bufferedWriter.write(lst + COMMA + result + ENTER);
+
+                if (supplyResult == 0) {
+                    supplyResult = result;
+                } else {
+                    int difference = supplyResult - result;
+                    bufferedWriter.write("result," + difference);
+                }
+            }
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+    }
```
please separate this part of code into new method

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,72 @@
+                list.add(rows);
+            }
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
```
bad practice, this message don't have enough information about the Exception

ChumakAn commented on src/main/java/core/basesyntax/WorkWithFile.java:
```diff
@@ -1,8 +1,72 @@
+        try {
+            BufferedReader bufferedReader = new BufferedReader(new FileReader(fromFile));
+            while (bufferedReader.ready()) {
+                String[] rows = bufferedReader.readLine().split(",");
```
you have a COMMA constant, why don't you use it here?
